### PR TITLE
Add Please Run That

### PR DESCRIPTION
**Pull request by Sebicodes99**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.sebicodes.Prt.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+app-id: org.sebicodes.Prt
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+tag: "1.0.0"
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:5]:
```suggestion
```

File: org.sebicodes.Prt.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+commit": "d60a7d3ee075d4c74f6a8120f68fb689e48ea419"
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:6]:
```suggestion
```

File: org.sebicodes.Prt.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+command: runner.sh
+modules:
+  - name: app
+    buildsystem: simple
+    build-commands:
+      - install -D runner.sh /app/bin/runner.sh
+      - install -D app.py /app/bin/app.py
+      - install -D .desktop /app/share/applications/com.sebicodes.Prt.desktop
+      - install -D plruth.png /app/share/icons/hicolor/500x500/apps/com.sebicodes.Prt.png
+    sources:
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:16]: this look like you should have a single source of type "git" instead.

File: org.sebicodes.Prt.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+commit": "d60a7d3ee075d4c74f6a8120f68fb689e48ea419"
+command: runner.sh
+modules:
+  - name: app
+    buildsystem: simple
+    build-commands:
+      - install -D runner.sh /app/bin/runner.sh
+      - install -D app.py /app/bin/app.py
+      - install -D .desktop /app/share/applications/com.sebicodes.Prt.desktop
+      - install -D plruth.png /app/share/icons/hicolor/500x500/apps/com.sebicodes.Prt.png
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:15]: 500x500 is not a valid icon size.

Comment (hfiguiere) [reply org.sebicodes.Prt.yml:15]: also the name doesn't match the id.

File: org.sebicodes.Prt.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
+app-id: org.sebicodes.Prt
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:1]: `sebicodes.org` doesn't exist. This app-id doesn't meet the requirements.

File: org.sebicodes.Prt.yml (hunk 4 above)
Comment (Sebicodes99) [reply org.sebicodes.Prt.yml:13]: Hello, thanks for your revision. What should be a valid size?
I will fix it now

File: org.sebicodes.Prt.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+       - type: archive
+         path: https://raw.githubusercontent.com/Sebicodes99/pleaserunthat-beta/stable/Flatpak/runner.sh
+         sha256: a333c84a33aa7671be1adf8ea8d8ee067ae922adfc7f08683d19830cffbda838
+       - type: archive
+         path: https://github.com/Sebicodes99/pleaserunthat-beta/blob/stable/Flatpak/plruth.png?raw=true
+         sha256: 06b636ef882633bf779158488df8b396f91b06c3d2401078648d201ef750559a
+       - type: archive
+         path: https://raw.githubusercontent.com/Sebicodes99/pleaserunthat-beta/stable/Flatpak/org.sebicodes.Prt.appdata.xml
+         sha256: 3e8d8375df499c97561678b73f0c92556268ec321c537d9c53438e81e531c1c9
+finish-args:
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:30]: finish args is usually above `modules`.

also the app doesn't display any GUI?

Comment (Sebicodes99) [reply org.sebicodes.Prt.yml:30]: No, It's a terminal app. .desktop includes it.

Comment (Sebicodes99) [reply org.sebicodes.Prt.yml:30]: I will change it

Comment (Sebicodes99) [reply org.sebicodes.Prt.yml:30]: Done

Comment (hfiguiere) [reply org.sebicodes.Prt.yml:30]: CLI apps really aren't the best use case for flatpak and flathub. Just setting the expectations here.

File: org.sebicodes.Prt.yml (hunk 5 above)
Comment (Sebicodes99) [reply org.sebicodes.Prt.yml:1]: But I don't have any domain, what should i do?

File: org.sebicodes.Prt.yml (hunk 3 above)
Comment (Sebicodes99) [reply org.sebicodes.Prt.yml:16]: and I should clone the entire repo instead?

File: org.sebicodes.Prt.yml (hunk 5 above)
Comment (hfiguiere) [reply org.sebicodes.Prt.yml:1]: use one associated to the code hosting https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: org.sebicodes.Prt.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    buildsystem: simple
+    build-commands:
+      - install -D runner.sh /app/bin/runner.sh
+      - install -D app.py /app/bin/app.py
+      - install -D .desktop /app/share/applications/org.sebicodes.Prt.desktop
+      - install -D plruth.png /app/share/icons/hicolor/500x500/apps/org.sebicodes.Prt.png
+    sources:
+      - type: git
+        url: https://github.com/Sebicodes99/pleaserunthat-beta.git
+        branch: stable
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:19]: it should be a commit or a tag. NEVER a branch.

File: org.sebicodes.Prt.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+app-id: org.sebicodes.Prt
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: runner.sh
+finish-args:
+  - --share=network
+modules:
+  - name: app
```
Comment (hfiguiere) [inline org.sebicodes.Prt.yml:9]: it's usualla a practice to use a module name that relate to the application. not something so generic.

Comment (Sebicodes99) [reply org.sebicodes.Prt.yml]: FIxed. Is it right now?

File: org.sebicodes.Prt.prt.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+app-id: io.github.Prt.prt
```
Comment (hfiguiere) [inline org.sebicodes.Prt.prt.yml:1]: You don't seem to be user `prt`.

File: org.sebicodes.Prt.prt.yml [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+finish-args:
+  - --share=network
+modules:
+  - name: PleaseRunThat
+    buildsystem: simple
+    build-commands:
+      - install -D flatpak/runner.sh /app/bin/runner.sh
+      - install -D flatpak/app.py /app/bin/app.py
+      - install -D flatpak/.desktop /app/share/applications/org.sebicodes.Prt.desktop
+      - install -D flatpak/plruth.png /app/share/icons/hicolor/128x128/apps/org.sebicodes.Prt.prt.png
```
Comment (hfiguiere) [inline org.sebicodes.Prt.prt.yml:15]: the image isn't 128x128.

File: org.sebicodes.Prt.prt.yml (hunk 9 above)
Comment (Sebicodes99) [reply org.sebicodes.Prt.prt.yml:1]: oh, so prt has to be my username or Prt?

Comment (hfiguiere) [reply org.sebicodes.Prt.prt.yml]: it's `io.github.` ORGINAZATION/USER `.` APPNAME

File: org.sebicodes.Prt.prt.yml (hunk 10 above)
Comment (Sebicodes99) [reply org.sebicodes.Prt.prt.yml]: Done

**Comment by Sebicodes99** [comment]
Fixed every problem, now the manifest should be right.

File: io.github.Prt.Sebicodes99.yml [hunk 11]
```diff
@@ -0,0 +1,19 @@
+app-id: io.github.Prt.Sebicodes99
```
Comment (hfiguiere) [inline io.github.Prt.Sebicodes99.yml:1]: As I said, you are not user `Prt`.

Comment (Sebicodes99) [reply io.github.Prt.Sebicodes99.yml:1]: Oh sorry, I tought it was the lowercase one, I will fix it

**Comment by Sebicodes99** [comment]
Fixed everything.

**Comment by hfiguiere** [comment]
pretty sure the build will fail

bot, build io.github.Sebicodes99.Prt

**Comment by hfiguiere** [comment]
`install: cannot stat 'flatpak/runner.sh': No such file or directory`

file names are case sensitive.

**Comment by Sebicodes99** [comment]
Done! Fixed it

File: io.github.Sebicodes99.Prt.yml [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+sdk: org.freedesktop.Sdk
+command: runner.sh
+finish-args:
+  - --share=network
+modules:
+  - name: PleaseRunThat
+    buildsystem: simple
+    build-commands:
+      - install -D Flatpak/runner.sh /app/bin/runner.sh
+      - install -D Flatpak/app.py /app/bin/app.py
```
Comment (hfiguiere) [inline io.github.Sebicodes99.Prt.yml:13]: actually this file is not in this locatin.

**Comment by hfiguiere** [comment]
also there is no appstream file installed. It's mandatory.

**Comment by Sebicodes99** [comment]
Where should it be installed?

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
oh, I forgot to change tag

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by hfiguiere** [comment]
The icon name in the desktop file is invalid. It should be just the icon name. No path. no extension.

Also the exec should just be the executable name (with extension) a not a path.

and as a side note, not blocking this, naming in `.desktop` is just the wrong idea because it's invalid and end up being "hidden"

File: io.github.Sebicodes99.Prt.yml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    build-commands:
+      - install -D Flatpak/runner.sh /app/bin/runner.sh
+      - install -D app.py /app/bin/app.py
+      - install -D Flatpak/.desktop /app/share/applications/io.github.Sebicodes99.Prt.desktop
+      - install -D Flatpak/plruth.png /app/share/icons/hicolor/128x128/apps/io.github.Sebicodes99.Prt.png
+      - install -D Flatpak/io.github.Sebicodes99.Prt.appdata.xml /app/share/metainfo/io.github.Sebicodes99.Prt.appdata.xml
+    sources:
+      - type: git
+        url: https://github.com/Sebicodes99/pleaserunthat-beta.git
+        tag: 1.6.0
```
Comment (hfiguiere) [inline io.github.Sebicodes99.Prt.yml:20]: this tag doesn't exist either.

**Comment by hfiguiere** [comment]
also appstream file is missing mandory `content_rating`, `screenshots` and `releases`

**Comment by hfiguiere** [comment]
Also in the appstream file the type should be `console-application` since it is a console application.

File: io.github.Sebicodes99.Prt.yml (hunk 13 above)
Comment (Sebicodes99) [reply io.github.Sebicodes99.Prt.yml:20]: Because I was just launching it, now it exists.

**Comment by Sebicodes99** [comment]
The new release added The first 3, is the screenshot size good or will it give an error?

**Comment by Sebicodes99** [comment]
Fixed appstream and .desktop

**Comment by Sebicodes99** [comment]
What should I name .desktop?

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by hfiguiere** [comment]
```
Error loading AppData file: failed to parse /srv/buildbot/worker/build-x86_64-7/build/.flatpak-builder/rofiles/rofiles-0JD2HR/files/share/appdata/io.github.Sebicodes99.Prt.appdata.xml: Error on line 36 char 13: Element “component” was closed, but the currently open element is “releases”
```

**Comment by Sebicodes99** [comment]
> ```
> Error loading AppData file: failed to parse /srv/buildbot/worker/build-x86_64-7/build/.flatpak-builder/rofiles/rofiles-0JD2HR/files/share/appdata/io.github.Sebicodes99.Prt.appdata.xml: Error on line 36 char 13: Element “component” was closed, but the currently open element is “releases”
> ```

Fixed!

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
Bot gave me this error: <metadata_license> is not valid [GNU General Public License v3.0]
What should I write instead?

**Comment by hfiguiere** [comment]
a SPDX license identifier is expected

https://spdx.dev/licenses/

**Comment by Sebicodes99** [comment]
Thanks but, Before this, I used it but it told me the same, what should i do?

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by hfiguiere** [comment]
oh, metadata license GPL is not ok. It should be CC0-1.0 or a few other non copyleft license. It's is just for the appstream, not the project.

**Comment by Sebicodes99** [comment]
Oh, Thanks

**Comment by Sebicodes99** [comment]
I will change it

**Comment by Sebicodes99** [comment]
bot, build io.github.Sebicodes99.Prt

**Comment by Sebicodes99** [comment]
Thanks a lot, now that works

**Comment by Sebicodes99** [comment]
It's already built

**Comment by Sebicodes99** [comment]
Now, I'm ready for publishing. ¡Thanks a lot for your help!

**Comment by Sebicodes99** [comment]
Hello, Trying to change the name of the branch closed the pull request so I restored it.

**Comment by barthalion** [comment]
I think there should be some Tk-based GUI or at least a proper field validation. Right now it accepts anything and happily fails; and it's neither TUI nor CLI so it's kinda weird thing to publish to Flathub.

**Comment by barthalion** [comment]
The alternative would be to drop the desktop file and mark it as a console application in metainfo.

**Comment by Sebicodes99** [comment]
But, the idea is to make it fast to run. It's already marked as a console application.

**Pull request closed without merging**